### PR TITLE
Support generating OpenAPI operation and associated fields

### PR DESCRIPTION
**Pull request by captainsafia**
Standard implementation of summary, description, and tags within an OpenAPI operation using endpoint metadata.

### CODE REVIEW

File: src/OpenApi/src/Helpers/OpenApiTagComparer.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        if (x is null && y is null)
+        {
+            return true;
+        }
+        if (x is null || y is null)
+        {
+            return false;
+        }
+        return x.Name == y.Name;
```
Comment (martincostello) [inline src/OpenApi/src/Helpers/OpenApiTagComparer.cs:26]: Should this be case sensitive or insensitive? I wonder if using `string.Equals()` with the appropriate ordinal comparison would make the intent clearer in either case?

Comment (captainsafia) [reply src/OpenApi/src/Helpers/OpenApiTagComparer.cs:26]: Good question! The OpenAPI specification states that all field _names_ should be considered case sensitive ([ref](https://spec.openapis.org/oas/v3.1.0#format)), but it doesn't provide an opinion on field _values_.  For consistencies sake, I think it makes sense to apply the same sensitivity rules to values as well.

I'll pop a comment in this code clarifying why this case sensitive comparison is used.

File: src/OpenApi/src/Services/OpenApiDocumentService.cs [hunk 2]
```diff
@@ -19,12 +20,18 @@ internal sealed class OpenApiDocumentService(
...
+    // Note: internal for testing.
+    internal readonly HashSet<OpenApiTag> _capturedTags = new(OpenApiTagComparer.Instance);
+
     public Task<OpenApiDocument> GetOpenApiDocumentAsync()
     {
         var document = new OpenApiDocument
         {
             Info = GetOpenApiInfo(),
-            Paths = GetOpenApiPaths()
+            Paths = GetOpenApiPaths(),
```
Comment (amcasey) [inline src/OpenApi/src/Services/OpenApiDocumentService.cs:33]: This is basically just a style preference, but I would find it more readable if the set were local to `GetOpenApiDocumentAsync` and threaded through as a parameter.  That would make it clear, for example, that there are no thread safety considerations and that the set is recomputed (idempotently, at the moment) on each call to `GetOpenApiDocumentAsync` (maybe there's only one?).

**Review by amcasey — APPROVED** [review]

File: src/OpenApi/src/Services/OpenApiDocumentService.cs (hunk 2 above)
Comment (captainsafia) [reply src/OpenApi/src/Services/OpenApiDocumentService.cs]: > (idempotently, at the moment) ... (maybe there's only one?).

Good note! Yes, the tags are recomputed idempotently on each `GetOpenApiDocumentAsync` invocation. `GetOpenApiDocumentAsync` is currently called every time the user hits the `/openapi/{documentName}.json` endpoint. It is a little harder to observe this the way it's currently structured so I'll adjust.

**Pull request merged**